actionview/lib/action_view/record_identifier.rb

```diff
@@ -94,6 +95,39 @@ def dom_id(record, prefix = nil)
       end
     end
 
+    # If you need to prepend multiple prefixes at a time, pass them as an Array
+    # or as arguments:
+    #
+    #   dom_ids(Post.find(45), :edit, custom: true)                # => "edit_post_45 custom_post_45"
+    #   dom_ids(Post.new, :edit, custom: true)                     # => "edit_post custom_post"
+    #
+    #   dom_ids(Post.find(45), [:edit, { custom: true }, :title])  # => "edit_post_45 custom_post_45 title_post_45"
+    #   dom_ids(Post.new, [:edit, { custom: true }, :title])       # => "edit_post custom_post title_post"
+    #
+    # For example, reference multiple elements from within an aria-labelledby
+    # attribute:
+    #
+    #   tag.button "Edit", id: dom_id(@post, :edit), aria: { labelledby: dom_ids(@post, [:edit, { custom: true }, :title] }
+    #   # => <button id="edit_post_1" aria-labelledby="edit_post_1 custom_post_1 title_post_1">Edit</button>
+    def dom_ids(record, *prefixes, **conditional_prefixes)
+      prefixes = (Array.wrap(prefixes) + [conditional_prefixes].compact_blank).flatten
```

Comment:
Another possible variation on this interface would declare the `prefixes:` keyword argument, which would require an `Array` argument, which would make this flattening unnecessary:

```ruby
def dom_ids(record, prefixes: [])
  # ...
end
```